web/packages/teleterm/src/ui/Search/SearchContext.tsx

```diff
@@ -158,6 +168,22 @@ export const SearchContextProvider: FC = props => {
     [isUserInteractionPaused]
   );
 
+  /**
+   * makeEventListener is similar to addWindowEventListener but meant for situations where you want
+   * to add a listener to an element directly. By wrapping the listener in makeEventListener, you
+   * make sure that the listener will be removed when the interaction with the search bar is paused.
+   */
+  const makeEventListener = useCallback(
```

Comment:
Technically the function has nothing to do with event listeners, it just simply does not return the passed value if `isUserInteractionPaused` is true.

So we could have a more generic name but since this function would be used only for event listeners, I feel like `makeEventListener` is a better name than a generic one.

I tried coming up with something like `makePausableEventListener` but I didn't think of anything that I'd like.